app/Repositories/Doctrine/AccountRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repositories\Doctrine;
+
+use App\Database\Entities\Account;
+use App\Database\Exceptions\InvalidTypeSubscriptionException;
+use App\Repositories\Doctrine\ORM\AbstractRepository;
+use App\Repositories\Interfaces\AccountRepositoryInterface;
+
+final class AccountRepository extends AbstractRepository implements AccountRepositoryInterface
+{
+    /**
+     * Find by subscription type.
+     *
+     * @param string $subscriptionType
+     *
+     * @return mixed[]
+     */
+    public function findBySubscriptionType(string $subscriptionType): array
+    {
+        try {
+            if ($subscriptionType !== 'monthly' && $subscriptionType !== 'lifetime') {
```

Comment:
I suggest to store  subscription types as constants in the entity

/** string */
public const SUBSCRIPTION_TYPE_MONTHLY = 'monthly';

/** string[] */
public const SUBSCRIPTION_TYPES = [
  self::SUBSCRIPTION_TYPE_MONTHLY
  ... 
];